File: D2L1/RainyRoad.hs
module RainyRoad where 

toString :: Bool -> String
toString False = "NO"
toString True  = "YES"

passable :: (Char, Char) -> Bool
passable ('W', 'W') = False
passable _ = True

isReachable :: [String] -> String
isReachable road = toString $ all passable $ zip (road!!0) (road!!1)

File: D2L1/RainyRoad.cpp
#include <vector>
#include <string>
using namespace std;

string isReachable(vector<string> const &road) {
    static const string NO = "NO";
    static const string YES = "YES";

    int n = road[0].size();
    for (int i = 1; i != n; ++i) {
        if (road[0][i] == 'W' && road[1][i] == 'W') {
            return NO;
        } 
    }
    return YES;
}
